Code/Framework/AzCore/AzCore/Settings/SettingsRegistryImpl.cpp

```diff
@@ -217,18 +216,43 @@ namespace AZ
     {
         NotifyEventHandler notifyHandler{ AZStd::move(callback) };
         {
-            AZStd::scoped_lock lock(m_settingMutex);
+            AZStd::scoped_lock lock(m_notifierMutex);
             notifyHandler.Connect(m_notifiers);
         }
         return notifyHandler;
     }
 
     void SettingsRegistryImpl::ClearNotifiers()
     {
-        AZStd::scoped_lock lock(m_settingMutex);
+        AZStd::scoped_lock lock(m_notifierMutex);
         m_notifiers.DisconnectAllHandlers();
     }
 
+    void SettingsRegistryImpl::SignalNotifier(AZStd::string_view jsonPath, Type type)
+    {
+        // Move the Notifier AZ::Event to a local AZ::Event in order to allow
+        // the notifier handlers to be signaled outside of the notifier mutex
+        // This allows other threads to register notifiers while this thread
+        // is invoking the handlers
```

Comment:
Note: seems like the idea of a thread safe AZ::Event needs to be revisited again. Would be better imho to have this logic captured in a thread safe AZ::Event instead of partly here and partly in the normal AZ::Event (the StealHandlers call).

Reply:
Not only should thread safety be a talking point, but the ability to asynchronously launch the handlers in parallel should also be one as well.

In the SignalHandlers function in the Settings Registry, the registered notifies could safely run in parallel.